rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/StorageActionsTests.scala

```diff
@@ -150,6 +150,58 @@ trait StorageActionsTests[F[_]]
     } yield ()
   }
 
+  "producing and then consuming on the same channel with peek" should
+    "return the continuation and data and not remove the peeked data" in fixture {
+    (store, _, space) =>
+      val channel = "ch1"
+      val key     = List(channel)
+
+      for {
+        r1 <- space.produce(channel, "datum", persist = false)
+        d1 <- store.getData(channel)
+        _  = d1 shouldBe List(Datum.create(channel, "datum", false))
+        c1 <- store.getContinuations(key)
+        _  = c1 shouldBe Nil
+        _  = r1 shouldBe None
+
+        r2            <- space.consume(key, List(Wildcard), new StringsCaptor, persist = false, peeks = Set(0))
```

Comment:
This always tests peek on the first channel. Perhaps it would make sense to write a generative spec in which the peeked channel would be chosen randomly?

Reply:
yep, thinking how to build one